code/controllers/subsystems.dm

```diff
@@ -0,0 +1,46 @@
+#define NEW_SS_GLOBAL(varname) if(varname != src){if(istype(varname)){Recover();qdel(varname);}varname = src;}
+
+/datum/subsystem
```

Comment:
I'm still not quite sure why bay made the decision to implement this on top of the process scheduler. A process should be able to handle anything here just as easily.